### ISSUE TITLE
TST: `test_hermitian_modes` test failure

### ISSUE THREAD

**Issue by alugowski** [scipy/scipy#18736]
This test in the "Linux Meson tests / Meson build (3.11, false)" workflow has been failing on most recent CI runs. 

The test passes on the Python 3.10 workflow, and nearly always (but not always) fails on the Python 3.11 workflow. The difference between the two is that the 3.11 workflow installs dependencies from git instead of from PyPI. I went ahead and bisected the differences, and found that the test fails if numpy is installed from git but passes if numpy is installed from PyPI:

![Screenshot 2023-06-24 at 12 56 55 AM](https://github.com/scipy/scipy/assets/2730364/fcab5b63-53ba-4c59-bc01-58f2307b460a)

I ran several test runs of each because sometimes the test passes (even on the git version of numpy).

In other words, the test passes with this:
```
python -m pip install numpy
```
and fails with this:
```
python -m pip install git+https://github.com/numpy/numpy.git
```

Note that this installs the main branch of numpy, which is currently at 2.0.0dev0 with active development.

My contribution to fixing this is identifying this direction. I hope someone more familiar with the arpack code and/or numpy can track it down further.

**Comment by alugowski** [comment]
Note that pip can install a particular commit: `python -m pip install git+https://github.com/numpy/numpy.git@abcdefg`. This could help bisect it further.

**Comment by steppi** [comment]
Thanks @alugowski! This is very helpful. I'm still unable to reproduce this locally when installing numpy with pip, so it seems like there must be something more to it.

**Comment by alugowski** [comment]
> there must be something more to it

Indeed. Notice that the test also passes on the prerelease workflow, and that installs numpy nightly via conda. If it was a simple code issue then it would fail there too.

It must be something triggered by system config, like compilers or build flags? I think the best next step is to replicate the environment in docker. `act` may or may not help. I'm on ARM macOS and these x86 things get painfully slow so I can't help much. Alternatively run it in your own private GitHub Actions like I did (comment out the lines that restrict running the .yml to just the scipy/scipy repo)

**Comment by steppi** [comment]
> > there must be something more to it
> 
> Indeed. Notice that the test also passes on the prerelease workflow, and that installs numpy nightly via conda. If it was a simple code issue then it would fail there too.
> 
> It must be something triggered by system config, like compilers or build flags? I think the best next step is to replicate the environment in docker. `act` may or may not help. I'm on ARM macOS and these x86 things get painfully slow so I can't help much. Alternatively run it in your own private GitHub Actions like I did (comment out the lines that restrict running the .yml to just the scipy/scipy repo)

Yeah. I plan replicate the environment but it’ll be a few days before I have enough bandwidth.

**Comment by steppi** [comment]
> > > there must be something more to it
> > 
> > 
> > Indeed. Notice that the test also passes on the prerelease workflow, and that installs numpy nightly via conda. If it was a simple code issue then it would fail there too.
> > It must be something triggered by system config, like compilers or build flags? I think the best next step is to replicate the environment in docker. `act` may or may not help. I'm on ARM macOS and these x86 things get painfully slow so I can't help much. Alternatively run it in your own private GitHub Actions like I did (comment out the lines that restrict running the .yml to just the scipy/scipy repo)
> 
> Yeah. I plan replicate the environment but it’ll be a few days before I have enough bandwidth. I have a machine running Ubuntu 22.04 on X86_64 so no concerns with docker being slow.

**Comment by ilayn** [comment]
In the meantime, I'll relax the tolerances since this is getting a bit annoying just at the wake of a release.

**Comment by ilayn** [comment]
Done
